src/flag_gems/ops/cummin.py

```diff
@@ -295,6 +336,90 @@ def scan_then_fan(inp, out, out_indices, A, B, C, dtype):
             )
 
 
+@libentry()
+@triton.jit()
+def scan_part_min_abc_loop_kernel(
+    inp,
+    out,
+    out_indices,
+    B,
+    C,
+    loop_num,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid_a = tle.program_id(0)
+    pid_c = tle.program_id(1)
+
+    a_idx = pid_a
+    c_idx = pid_c
+    t_idx = tl.arange(0, BLOCK_SIZE)
+    ac_offset = a_idx * B * C + c_idx
+
+    # init
+    max_value = get_dtype_max(inp.type.element_ty)
+    prev_min_val = tl.full([], max_value, dtype=tl.float32)
+    prev_min_val_idx = tl.full([], 0, dtype=tl.int64)
+    last_mask = t_idx == (BLOCK_SIZE - 1)
+
+    for l_idx in tl.range(loop_num):
+        b_idx = l_idx * BLOCK_SIZE + t_idx
+        mask = b_idx < B
+        offset = ac_offset + b_idx * C
+
+        inp_vals = tl.load(inp + offset, mask=mask, other=max_value)
+        if (
+            tl.constexpr(inp_vals.dtype.is_int64())
+            or tl.constexpr(inp_vals.dtype.is_uint64())
+        ) or tl.constexpr(inp_vals.dtype.is_fp64()):
+            vals = inp_vals
+        elif tl.constexpr(inp_vals.dtype.is_int()):
+            vals = inp_vals.to(tl.int32)
+        else:
+            vals = inp_vals.to(tl.float32)
+        idxs = b_idx
+
+        # cummin
+        result, cummin_indices = tl_cummin(vals, idxs, axis=0)
+
+        # broadcast
+        prev_min_val_b = tl.broadcast_to(prev_min_val, (BLOCK_SIZE,))
+        prev_min_val_idx_b = tl.broadcast_to(prev_min_val_idx, (BLOCK_SIZE,))
+
+        # update result from prev val and idx
+        cummin_indices = tl.where(
+            result <= prev_min_val_b, cummin_indices, prev_min_val_idx_b
+        )
+        result = tl.minimum(result, prev_min_val_b)
+
+        # update global min val and idx
+        prev_min_val = tl.sum(tl.where(last_mask, result, 0.0), axis=0)
+        prev_min_val_idx = tl.sum(tl.where(last_mask, cummin_indices, 0), axis=0)
+
+        # store result
+        tl.store(out + offset, result, mask=mask)
+        tl.store(out_indices + offset, cummin_indices, mask=mask)
+
+
+def scan_then_fan_loop(inp, out, out_indices, A, B, C, dtype):
+    # TODO(all): tune on target board
+    BLOCK_SIZE = 1024
+    if B < 1024 * 4:
+        BLOCK_SIZE = triton.next_power_of_2(B)
+    loop_num = math.ceil(B / BLOCK_SIZE)
+
+    grid = (A, C)
+    with torch_device_fn.device(inp.device):
+        scan_part_min_abc_loop_kernel[grid](
+            inp,
+            out,
+            out_indices,
+            B,
+            C,
+            loop_num,
+            BLOCK_SIZE,
+        )
+
+
 def cummin(inp, dim=1, *, dtype=None):
```

Comment:
Actually, the interface should remain consistent with that of Aten for cummin as well. We should address this issue later.

Reply:
I've fix the inconsistent interface issue in the latest commit.
Can you help to review please?
Thanks!